src/axcrypt_fmt_plug.c

```diff
@@ -0,0 +1,354 @@
+    dyna_salt dsalt;
+    int version;
+    uint32_t key_wrapping_rounds;
+    unsigned char salt[16];
+    unsigned char wrappedkey[24];
+    char* keyfile;
+} *cur_salt;
+
+static void init(struct fmt_main *self)
+{
+#ifdef _OPENMP
+    int omp_t = 1;
+    omp_t = omp_get_max_threads();
+    self->params.min_keys_per_crypt *= omp_t;
+    omp_t *= OMP_SCALE;
+    self->params.max_keys_per_crypt *= omp_t;
+#endif
+    saved_key = mem_calloc(self->params.max_keys_per_crypt,
+                           sizeof(*saved_key));
+    any_cracked = 0;
+    cracked_size = sizeof(*cracked) * self->params.max_keys_per_crypt;
+    cracked = mem_calloc(cracked_size, 1);
+}
+
+static void done(void)
+{
+    MEM_FREE(cracked);
+    MEM_FREE(saved_key);
+}
+
+static int valid(char *ciphertext, struct fmt_main *self)
+{
+    char *p;
+    char *ctcopy;
+    char *keeptr;
+
+    if (strncmp(ciphertext, "$axcrypt$*", 10) != 0)
+        return 0;
+
+    ctcopy = strdup(ciphertext);
+    keeptr = ctcopy;
+    ctcopy += 10;        /* skip over "$axcrypt$*" */
+    if ((p = strtokm(ctcopy, "*")) == NULL)    /* version */
+        goto err;
+    if (!isdec(p))
+        goto err;
+    if (!atoi(p))
+        goto err;
+    if ((p = strtokm(NULL, "*")) == NULL)    /* iterations */
+        goto err;
+    if (!isdec(p))
+        goto err;
+    if (!atoi(p))
+        goto err;
+    if ((p = strtokm(NULL, "*")) == NULL)    /* salt */
+        goto err;
+    if (strlen(p) != 32)
+        goto err;
+    if ((p = strtokm(NULL, "*")) == NULL)    /* wrappedkey */
+        goto err;
+    if (strlen(p) != 48 )
+        goto err;
+    /* optional key-file following */
+
+    MEM_FREE(keeptr);
+    return 1;
+err:
+    MEM_FREE(keeptr);
+    return 0;
+}
+
+static void *get_salt(char *ciphertext)
+{
+    char *ctcopy = strdup(ciphertext);
+    char *keeptr = ctcopy;
+    char *p;
+    int i;
+    static struct custom_salt cs;
+    static void *ptr;
+
+    cs.keyfile = NULL;
+    ctcopy += 10;    /* skip over "$axcrypt$*" */
+    p = strtokm(ctcopy, "*");
+    cs.version = atoi(p);
+
+    p = strtokm(NULL, "*");
+    cs.key_wrapping_rounds = (uint32_t) atoi(p);
+
+    p = strtokm(NULL, "*");
+    for (i = 0; i < 16; i++)
+        cs.salt[i] = atoi16[ARCH_INDEX(p[i * 2])] * 16
+            + atoi16[ARCH_INDEX(p[i * 2 + 1])];
+
+    p = strtokm(NULL, "*");
+    for (i = 0; i < 24; i++)
+        cs.wrappedkey[i] = atoi16[ARCH_INDEX(p[i * 2])] * 16
+            + atoi16[ARCH_INDEX(p[i * 2 + 1])];
+
+    /* if key-file present */
+    if ((p = strtokm(NULL, "*")) != NULL){
+        cs.keyfile = (char*) mem_calloc(strlen(p)/2, sizeof(char));
+        for (i = 0; i < strlen(p)/2; i++)
+            cs.keyfile[i] = atoi16[ARCH_INDEX(p[i * 2])] * 16
+                + atoi16[ARCH_INDEX(p[i * 2 + 1])];
+    }
+    MEM_FREE(keeptr);
+
+    cs.dsalt.salt_cmp_offset = SALT_CMP_OFF(struct custom_salt, salt);
+    cs.dsalt.salt_cmp_size = SALT_CMP_SIZE(struct custom_salt, salt, wrappedkey, 0);
```

Comment:
I think this should be `SALT_CMP_SIZE(struct custom_salt, salt, keyfile, size_of_keyfile);` for really proper salt compares... @jfoug?

However, that would not be the cause of problem seen.